### ISSUE TITLE
Keys() method's return statement in interpreter.go does not appear to return ( keys []string)

### ISSUE THREAD

**Issue by kkruups** [d4l3k/go-pry#10]
Hi D413k,

If I read your code correctly you Keys method (in interpreter.go file) does not return a (keys []string) in body of method.
Please review accordingly  (see excerpt below).

```go
// Keys returns all keys in scope
func (scope *Scope) Keys() (keys []string) {
	currentScope := scope
	for currentScope != nil {
		for k := range currentScope.Vals {
			keys = append(keys, k)
		}
		currentScope = scope.Parent
	}
	return
}
```

**Comment by d4l3k** [comment]
Are you commenting on the fact that I'm using a naked return? It's kind of bad practice, so if you want to submit a pull request to refactor it. I'd be happy to accept it.

Unless I misunderstand what you're talking about.

**Comment by kkruups** [comment]
Hi d4l3k,

Yes I am commenting on the naked return IMHO, naked returns reduce readability.

I am also wondering if that function is memory safe:  func (scope *Scope) Keys() (keys []string)

I ran a function equivalent of your method ,  testNakeReturn()  , see below.  and ran out of memory space. In my version of course I running an infinite loop since <s> is never nil.

Error Indicated  (It appears I ran out of memory space)
fatal error: runtime: cannot map pages in arena address space
 



func testNakedReturn() (keys []string) {
	s := []string{"abe", "babe", "cape"}

	for s != nil {
		for _, k:= range s {
			keys = append(keys, k)
			//fmt.Println(k)
		}
	}
	return
}
